### PR TITLE
feat: onboard fortinet firewall module

### PR DESCRIPTION
**Pull request by rling-equinix**

### CODE REVIEW

**Comment by kpdhulipala** [comment]
we need to add support to provide explicit wan/ssh interface selection for this and existing modules too.
image

File: modules/fortigate-firewall/variables.tf [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    condition     = var.additional_bandwidth == 0 || (var.additional_bandwidth >= 25 && var.additional_bandwidth <= 5001)
+    error_message = "Additional internet bandwidth should be between 25 and 5001 Mbps."
+  }
+}
+
+variable "ssh_key" {
+  description = "SSH public key for a device"
+  type        = object({
+    username = string
+    key_name = optional(string)
```
Comment (kpdhulipala) [inline modules/fortigate-firewall/variables.tf:121]: any reason to declare key_name as optional attribute?

File: modules/fortigate-firewall/variables.tf [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+variable "license_file" {
+  description = "Path to the device license file"
+  type        = string
+  default     = ""
+}
+
+variable "license_token" {
+  description = "License token"
+  type        = string
+  default     = ""
```
[the remark this answers is not in the document]
Comment (rling-equinix) [reply modules/fortigate-firewall/variables.tf]: Resolved.

File: examples/fg-vm-cluster/main.tf
[the remark this answers is not in the document]
Comment (rling-equinix) [reply examples/fg-vm-cluster/main.tf:1]: I think the existing example provides enough guidance for license_file, license_token and license_file_id.

File: modules/fortigate-firewall/variables.tf (hunk 1 above)
Comment (rling-equinix) [reply modules/fortigate-firewall/variables.tf]: Resolved.

**Review by kpdhulipala — APPROVED** [review]

**Review by schadalawada — APPROVED** [review]

**Pull request merged**